project/frontend_views/views.py

```diff
@@ -266,3 +267,20 @@ def about_view(request):
 
 def support_us_view(request):
     return TemplateResponse(request, 'static_templates/support_us.html', {})
+
+@csrf_exempt
+def civi2csv(request):
+    import csv
+    if request.method == 'POST':
```

Comment:
Try removing this line to see if it simplifies the client code. All we should need on the client with an href, which makes a GET request. Checking for POST may be over-complicating this solution.